File: backend/app/schemas/page.py
import uuid
from datetime import datetime

from pydantic import BaseModel, Field


class PageCreate(BaseModel):
    site_id: uuid.UUID
    url: str = Field(min_length=1, max_length=2048)


class PageOut(BaseModel):
    id: uuid.UUID
    site_id: uuid.UUID
    url: str
    title: str | None
    meta_description: str | None
    summary: str | None
    content_text: str | None
    hero_image_url: str | None
    key_points: list[str] | None
    keywords: list[str] | None
    status: str
    last_crawled_at: datetime | None
    created_at: datetime
    updated_at: datetime
    posts_count: int

    model_config = {"from_attributes": True}


class PageListResponse(BaseModel):
    items: list[PageOut]
    total: int
    page: int
    limit: int


class PagePostSummary(BaseModel):
    id: uuid.UUID
    platform: str
    status: str
    title: str | None
    created_at: datetime

    model_config = {"from_attributes": True}


class PagePostListResponse(BaseModel):
    items: list[PagePostSummary]

File: frontend/src/types/page.ts
export type PageStatus = 'pending' | 'crawled' | 'failed'

export interface Page {
  id: string
  site_id: string
  url: string
  title: string | null
  meta_description: string | null
  summary: string | null
  content_text: string | null
  hero_image_url: string | null
  key_points: string[] | null
  keywords: string[] | null
  status: PageStatus
  last_crawled_at: string | null
  created_at: string
  updated_at: string
  posts_count: number
}

export interface PageListResponse {
  items: Page[]
  total: number
  page: number
  limit: number
}

export interface PageCreateInput {
  site_id: string
  url: string
}
